### PR TITLE
[python] ensure adhoc arch command uses target arch

### PR DESCRIPTION
**Pull request by sjkelly**

### CODE REVIEW

File: P/Python/build_tarballs.jl [hunk 1]
```diff
@@ -22,7 +22,11 @@ apk add zlib-dev libffi-dev
 
 # Create fake `arch` command:
 echo '#!/bin/bash' >> /usr/bin/arch
-echo 'echo i386'   >> /usr/bin/arch
+if [[ "${target}" == *-apple-* ]]; then
+    echo `echo i386`  >> /usr/bin/arch
```
Comment (giordano) [inline P/Python/build_tarballs.jl:26]: I don't think this is correct

Comment (sjkelly) [reply P/Python/build_tarballs.jl:26]: Doesn't appear to be either :/

Comment (sjkelly) [reply P/Python/build_tarballs.jl:26]: back ticks...

File: P/Python/build_tarballs.jl [hunk 2]
```diff
@@ -22,7 +22,11 @@ apk add zlib-dev libffi-dev
 
 # Create fake `arch` command:
 echo '#!/bin/bash' >> /usr/bin/arch
-echo 'echo i386'   >> /usr/bin/arch
+if [[ "${target}" == *-apple-* ]]; then
+    echo 'echo i386'  >> /usr/bin/arch
+else
+    echo 'echo `echo $target | cut -d - -f 1`'  >> /usr/bin/arch
```
Comment (giordano) [inline P/Python/build_tarballs.jl:28]: If you use double quotes you can basically inline the result
```suggestion
    echo "echo $(echo $target | cut -d - -f 1)"  >> /usr/bin/arch
```
But why not always using `uname -m` instead?

Comment (sjkelly) [reply P/Python/build_tarballs.jl:28]: When I tried `uname -m` in the apple `runshell` (I presume this is reflective of the CI environment), it gives "x86_64" which is not what cmake wants.

**Comment by sjkelly** [comment]
Does this need a version bump?

**Comment by giordano** [comment]
> Does this need a version bump?

We try to keep the same version as upstream as far as possible

**Pull request merged**